src/QMCWaveFunctions/tests/test_DiracDeterminantBatched.cpp

```diff
@@ -689,7 +688,135 @@ void test_DiracDeterminantBatched_spinor_update(const int delay_rank, DetMatInve
+  }
+
+  std::fill(ratios.begin(), ratios.end(), 0);
+  std::fill(grads.begin(), grads.end(), 0);
+  dd.mw_ratioGradWithSpin(dd_ref_list, p_ref_list, 1, ratios, grads, spingrads);
+  for (int iw = 0; iw < grads.size(); iw++)
+  {
+    REQUIRE(ratios[iw] == ComplexApprox(ValueType(1.7472917722050971, 1.1900872950904169)));
+    REQUIRE(grads[iw][0] == ComplexApprox(ValueType(0.5496675534224996, -0.07968022499097227)));
+    REQUIRE(grads[iw][1] == ComplexApprox(ValueType(0.4927399293808675, -0.29971549854643653)));
+    REQUIRE(grads[iw][2] == ComplexApprox(ValueType(1.2792642963632226, 0.12110307514989149)));
+    REQUIRE(spingrads[iw] == ComplexApprox(ValueType(1.164708841479661, 0.9576425115390172)));
+  }
+
+  //reject move and check for initial values for mw_evalGrad
+  std::fill(grads.begin(), grads.end(), 0);
+  elec_.mw_accept_rejectMove<CoordsType::POS_SPIN>(p_ref_list, 1, {false, false});
+  dd.mw_evalGrad(dd_ref_list, p_ref_list, 1, grads);
+  for (int iw = 0; iw < grads.size(); iw++)
+  {
+    REQUIRE(grads[iw][0] == ComplexApprox(G_list[iw].get()[1][0]));
+    REQUIRE(grads[iw][1] == ComplexApprox(G_list[iw].get()[1][1]));
+    REQUIRE(grads[iw][2] == ComplexApprox(G_list[iw].get()[1][2]));
+  }
+
+  /* uncomment when mw_evalGradWithSpin is implemented
+  std::fill(grads.begin(), grads.end(), 0);
+  std::fill(spingrads.begin(), spingrads.end(), 0);
+  dd.mw_evalGradWithSpin(dd_ref_list, p_ref_list, 1, grads, spingrads);
+  for (int iw = 0; iw < grads.size(); iw++)
+  {
+    REQUIRE(grads[iw][0] == ComplexApprox(G_list[iw].get()[1][0]));
+    REQUIRE(grads[iw][1] == ComplexApprox(G_list[iw].get()[1][1]));
+    REQUIRE(grads[iw][2] == ComplexApprox(G_list[iw].get()[1][2]));
+    REQUIRE(spingrads[iw] == ComplexApprox(ValueType(1.18922259, 2.80414598)));
+  }
+  */
+
+  //now make and accept move, checking new values
+  elec_.mw_makeMove(p_ref_list, 1, displs);
+  elec_.mw_accept_rejectMove<CoordsType::POS_SPIN>(p_ref_list, 1, {true, true});
+
+  G  = 0;
+  L  = 0;
+  G2 = 0;
+  L2 = 0;
+  dd.mw_evaluateLog(dd_ref_list, p_ref_list, G_list, L_list);
+  for (int iw = 0; iw < dd_ref_list.size(); iw++)
+  {
+    PsiValueType ref = dd_ref_list[iw].getValue();
+    REQUIRE(std::log(ref) == ComplexApprox(ValueType(-0.41337396772929913, 1.4774106123071726)));
+    REQUIRE(G_list[iw].get()[1][0] == ComplexApprox(ValueType(0.5496675534224996, -0.07968022499097227)));
+    REQUIRE(G_list[iw].get()[1][1] == ComplexApprox(ValueType(0.4927399293808675, -0.29971549854643653)));
+    REQUIRE(G_list[iw].get()[1][2] == ComplexApprox(ValueType(1.2792642963632226, 0.12110307514989149)));
+  }
+
+  /* uncomment when mw_evalGradWithSpin is implemented
+  dd.mw_evalGradWithSpin(dd_ref_list, p_ref_list, 1, grads, spingrads);
+  for (int iw = 0; iw < grads.size(); iw++)
+    REQUIRE(spingrads[iw] == ComplexApprox(ValueType(1.164708841479661, 0.9576425115390172)));
+  */
```

Comment:
Still needed?

Reply:
Yea this should be uncommented when I add a mw_evalGradWithSpin to DiracDeterminantBatched. This only has   implemented mw_ratioGradWithSpin so that is all I test at the moment. I can remove it if desired and add it back in the next PR when I implement mw_evalGradWithSpin if desired